### PR TITLE
Implement cupy.sort.

### PR DESCRIPTION
**Pull request by takagi**
This PR implements cupy.sort, resending the changes of https://github.com/pfnet/chainer/pull/2539 and https://github.com/pfnet/chainer/pull/2581 for chainer v1.

### CODE REVIEW

**Comment by okuta** [comment]
LGTM except document.
Please add reference.

File: cupy/sorting/sort.py [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+
+    .. seealso:: :func:`numpy.sort`
+
+    """
+    ret = a.copy()
+    ret.sort()
+    return ret
+
 
 # TODO(okuta): Implement sort
```
Comment (okuta) [inline cupy/sorting/sort.py]: Please remove this line.

Comment (takagi) [inline cupy/sorting/sort.py]: I have removed the line.

**Comment by takagi** [comment]
> Please modify rst file.

You mean this change?
https://github.com/cupy/cupy/pull/55/files#diff-aa62295410f7a44cd462701d1c9b95fa

**Comment by okuta** [comment]
Sorry, I missed it.

**Comment by okuta** [comment]
OK, I will merge.

**Pull request merged**